xarray/core/common.py

```diff
@@ -823,3 +823,26 @@ def ones_like(other, dtype=None):
     """Shorthand for full_like(other, 1, dtype)
     """
     return full_like(other, 1, dtype)
+
+
+def is_np_datetime_like(dtype):
+    """Check if a dtype is a subclass of the numpy datetime types
+    """
+    return (np.issubdtype(dtype, np.datetime64) or
+            np.issubdtype(dtype, np.timedelta64))
+
+
+def contains_netcdftime_datetimes(var):
+    """Check if a variable contains netcdftime datetime objects"""
+    try:
+        from netcdftime._netcdftime import datetime
```

Comment:
can we use `from netcdftime import datetime` yet?

Reply:
With the standalone version of netcdftime, yes.  I stuck with `from netcdftime._netcdftime import datetime` for now because it works with both the standalone version and the version packaged with netCDF4.  Should I make this line version-dependent to hint that using the private API should eventually be phased out?

Reply:
you could probably use and/or extend my `_import_netcdftime` helper function:

https://github.com/pydata/xarray/blob/e1dc51572e971567fd3562db0e9f662e3de80898/xarray/coding/times.py#L41

Reply:
It would be better to have fallback path, e.g.,
```python
try:
    try:
        from netcdftime import datetime
    except ImportError:
        from netcdftime._netcdftime import datetime
except Import Error:
    # no netcdftime
```

Reply:
@shoyer the issue is though that with an older version of netcdftime importing `datetime` will succeed, but return the wrong result (https://github.com/Unidata/netcdftime/issues/8).

Reply:
OK, that's awful. So I guess we need to default to `from netcdftime._netcdftime import datetime` but we can at least add a comment noting why.

(I'm thinking that eventually `from netcdftime._netcdftime import datetime` may no longer work due to its use of private API, so it's better not to break when that happens.)

Reply:
Thanks @jhamman - following your example I gave writing an `_import_netcdftime_datetime` function a try.  It uses the private API pathway for versions of netcdftime before the netCDF4 split, and the public API pathway post split.

@shoyer it adds a bit a code, but should alleviate the risk of `from netcdftime._netcdftime import datetime` no longer working in the future.  I'm happy to revert back if you feel like this is unnecessarily complicated.